File: LeetCode-Problems/Easy/Number-Consistent-Strings.java
//Java
class Solution {
    public int countConsistentStrings(String allowed, String[] words) {
        int count = 0;
        Set<String> allowedChars = new HashSet<String>();
        for (int i = 0; i < allowed.length(); i++){
            allowedChars.add(allowed.substring(i, i+1));
        }
        for (int i = 0; i < words.length; i++){
            boolean isConsistent = true;
            for (int j = 0; j < words[i].length(); j++){
                if (!(allowedChars.contains(words[i].substring(j, j+1)))){
                    isConsistent = false;
                    break;
                } 
            }
            if (isConsistent){
                count++;
            }
        }
        return count;
    }
}

File: LeetCode-Problems/Easy/Number-Consistent-Strings.py
# Python
class Solution(object):
    def countConsistentStrings(self, allowed, words):
        """
        :type allowed: str
        :type words: List[str]
        :rtype: int
        """
        count = 0;
        allowedSet = set()
        for x in range(0, len(allowed)):
            allowedSet.add(allowed[x:x+1])
        for x in range (0, len(words)):
            isConsistent = True
            for y in range (0, len(words[x])):
                if (words[x][y:y+1] not in allowedSet):
                    isConsistent = False
                    break
                isConsistent = True
            if (isConsistent):
                count = count + 1
        return count
